src/configuration/CertificateDetailsModal.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  CodeBlock,
+  CodeBlockCode,
+  Modal,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { pki, util } from 'node-forge';
+import { FC } from 'react';
+
+interface CertificateDetailsModalProps {
+  isModalOpen: boolean;
+  certs: pki.Certificate[];
+  secretName: string;
+  pem: string;
+  onCloseModal: () => void;
+}
+
+const CertificateDetailsModal: FC<CertificateDetailsModalProps> = ({
+  isModalOpen,
+  certs,
+  secretName,
+  pem,
+  onCloseModal,
+}) => {
+  const formatCertIssuer = (crt: pki.Certificate) => {
+    let result = '';
+    for (let i = 0; i < crt.issuer.attributes.length; i++) {
+      if (crt.issuer.attributes[i].shortName) {
+        result += crt.issuer.attributes[i].shortName;
+        result += ' = ';
+        result += crt.issuer.attributes[i].value;
+        if (i !== crt.issuer.attributes.length - 1) {
+          result += ', ';
+        }
+      }
+    }
+    return result;
+  };
+
+  const formatCertSubject = (crt: pki.Certificate) => {
+    let result = '';
+    for (let i = 0; i < crt.subject.attributes.length; i++) {
+      if (crt.subject.attributes[i].shortName) {
+        result += crt.subject.attributes[i].shortName;
+        result += ' = ';
+        result += crt.subject.attributes[i].value;
+        if (i !== crt.subject.attributes.length - 1) {
+          result += ', ';
+        }
+      }
+    }
+    return result;
+  };
+
+  //https://datatracker.ietf.org/doc/html/rfc5280
+  const getExtPropContent = (ext: any, key: string) => {
+    let result = '';
+    if (typeof ext[key] === 'object') {
+      result += JSON.stringify(ext[key]);
+    } else {
+      result += ext[key].toString();
+    }
+
+    return result;
+  };
+
+  const getExtensionsList = (crt: pki.Certificate, order: number) => {
+    return [
+      crt.extensions.map((ext, index) => (
+        <>
+          <Split key={'extension' + index + 'order' + order}>
+            <SplitItem>{index + 1} -</SplitItem>
+            <SplitItem isFilled>
+              <b>{ext.name}</b>
+            </SplitItem>
+          </Split>
+          {[
+            Object.keys(ext).map(
+              (key) =>
+                key !== 'name' &&
+                key !== 'value' && (
+                  <Split hasGutter key={ext.name + key}>
+                    <SplitItem></SplitItem>
+                    <SplitItem>
+                      <b>{key}: </b>
+                    </SplitItem>
+                    <SplitItem isFilled>
+                      <code>{getExtPropContent(ext, key)}</code>
+                    </SplitItem>
+                  </Split>
+                ),
+            ),
+          ]}
+        </>
+      )),
+    ];
+  };
```

Comment:
This function must be a React component as it is producing JSX.

```typescript
interface ExtensionsListProps {
  crt: pki.Certificate;
  order: number;
}

const ExtensionsList: FC<ExtensionsListProps> = ({ crt, order }) => {
  //https://datatracker.ietf.org/doc/html/rfc5280
  const getExtPropContent = (ext: any, key: string) => {
    let result = '';
    if (typeof ext[key] === 'object') {
      result += JSON.stringify(ext[key]);
    } else {
      result += ext[key].toString();
    }

    return result;
  };
  return (
    <>
      {crt.extensions.map((ext, index) => (
        <>
          <Split key={'extension' + index + 'order' + order}>
            <SplitItem>{index + 1} -</SplitItem>
            <SplitItem isFilled>
              <b>{ext.name}</b>
            </SplitItem>
          </Split>
          {[
            Object.keys(ext).map(
              (key) =>
                key !== 'name' &&
                key !== 'value' && (
                  <Split hasGutter key={ext.name + key}>
                    <SplitItem></SplitItem>
                    <SplitItem>
                      <b>{key}: </b>
                    </SplitItem>
                    <SplitItem isFilled>
                      <code>{getExtPropContent(ext, key)}</code>
                    </SplitItem>
                  </Split>
                ),
            ),
          ]}
        </>
      ))}
    </>
  );
};
```

Reply:
It's overall better to extract as much as named components as possible instead of having a list too many "indentation" levels of components.

Here's a side good effect from this coding style:

![image](https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/assets/86971992/0b554ce7-1dbb-421f-8718-3cf896b6f164)

In the hierarchy of components we find our Extension and ExtensionList and ExtensionKey sub components with the parameters they got. Quite handy when debugging is needed :slightly_smiling_face: